arelle/plugin/validate/DBA/rules/fr.py

```diff
@@ -816,6 +743,77 @@ def rule_fr57(
+                  "one or more of the sub-items below must be indicated: "
+                  "Declaration obligations according to the declaration order, including especially the Criminal Code as well as tax, levy and subsidy legislation (audit) (ReportingResponsibilitiesAccordingToTheDanishExecutiveOrderOnApprovedAuditorsReportsEspeciallyTheCriminalCodeAndFiscalTaxAndSubsidyLegislationAudit)"
+                  "Declaration obligations according to the declaration order, including especially the company law or similar legislation laid down for the company (audit) (ReportingResponsibilitiesAccordingToTheDanishExecutiveOrderOnApprovedAuditorsReportsEspeciallyTheCompaniesActOrEquivalentLegislationThatTheCompanyIsSubjectToAudit)"
+                  "Declaration obligations according to the declaration order, including especially the legislation on financial reporting, including on bookkeeping and storage of accounting material (audit) (ReportingResponsibilitiesAccordingToTheDanishExecutiveOrderOnApprovedAuditorsReportsEspeciallyLegislationOnFinancialReportingIncludingAccountingAndStorageOfAccountingRecordsAudit)"
+                  "Declaration obligations according to the declaration order, including other matters in particular (revision (ReportingResponsibilitiesAccordingToTheDanishExecutiveOrderOnApprovedAuditorsReportsEspeciallyOtherMattersAudit)"),
+        )
+
+
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+)
+def rule_fr59(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation]:
+    """
+    DBA.FR59: When the annual report contains an audit report, which is when TypeOfAuditorAssistance = Revisionspåtegning / Auditor's report on audited financial statements, then the concept
+    arr:DescriptionOfQualificationsOfAuditedFinancialStatement must be filled in.
+    """
+    modelXbrl = val.modelXbrl
+    descriptonFacts = modelXbrl.factsByQname.get(pluginData.descriptionOfQualificationsOfAuditedFinancialStatementsQn)
+    indicatorFacts = []
+    if descriptonFacts is not None:
+        return
+    auditorFacts = modelXbrl.factsByQname.get(pluginData.typeOfAuditorAssistanceQn)
+    if auditorFacts is not None:
+        for aFact in auditorFacts:
+            if aFact.xValid >= VALID:
+                if aFact.xValue in [
+                    pluginData.auditedFinancialStatementsDanish,
+                    pluginData.auditedFinancialStatementsEnglish,
+                ]:
+                    indicatorFacts.append(aFact)
+        if len(indicatorFacts) > 0:
+            yield Validation.error(
+                codes='DBA.FR59',
+                msg=_("DescriptionOfQualificationsOfAuditedFinancialStatement must be tagged when {} is tagged with the value of {}").format(
+                    pluginData.typeOfAuditorAssistanceQn.localName,
+                    indicatorFacts[0].xValue),
+                modelObject=indicatorFacts[0])
```

Comment:
Minor note: The indentation on this line has been changed/is incorrect.

It’s best to commit formatting or code reordering changes separately from functional changes, as it helps make code reviews easier and faster.